dbt/include/trino/macros/catalog.sql

```diff
@@ -1,54 +1,65 @@
-    select * from (
-
-        (
-            with tables as (
-
-                select
-                    table_catalog as "table_database",
-                    table_schema as "table_schema",
-                    table_name as "table_name",
-                    table_type as "table_type",
-                    null as "table_owner"
-
-                from {{ information_schema }}.tables
-                where
-                    table_schema != 'information_schema'
-                    and
-                    table_schema in ('{{ schemas | join("','") | lower }}')
-
-            ),
-
-            columns as (
-
-                select
-                    table_catalog as "table_database",
-                    table_schema as "table_schema",
-                    table_name as "table_name",
-                    null as "table_comment",
-
-                    column_name as "column_name",
-                    ordinal_position as "column_index",
-                    data_type as "column_type",
-                    null as "column_comment"
-
-                from {{ information_schema }}.columns
-                where
-                    table_schema != 'information_schema'
-                    and
-                    table_schema in ('{{ schemas | join("','") | lower }}')
-
-            )
-
-            select *
-            from tables
-            join columns using ("table_database", "table_schema", "table_name")
-            order by "column_index"
+        with tables as (
+
+            select
+                table_catalog as "table_database",
+                table_schema as "table_schema",
+                table_name as "table_name",
+                table_type as "table_type",
+                null as "table_owner"
+
+            from {{ information_schema }}.tables
+            where
+                table_schema != 'information_schema'
+                and
+                table_schema in ('{{ schemas | join("','") | lower }}')
+
+        ),
+
+        columns as (
+
+            select
+                table_catalog as "table_database",
+                table_schema as "table_schema",
+                table_name as "table_name",
+
+                column_name as "column_name",
+                ordinal_position as "column_index",
+                data_type as "column_type",
+                comment as "column_comment"
+
+            from {{ information_schema }}.columns
+            where
+                table_schema != 'information_schema'
+                and
+                table_schema in ('{{ schemas | join("','") | lower }}')
+
+        ),
+
+        table_comment as (
+
+            select
+                catalog_name as "table_database",
+                schema_name as "table_schema",
+                table_name as "table_name",
+                comment as "table_comment"
+
+            from system.metadata.table_comments
+            where
+                catalog_name = '{{ database }}'
```

Comment:
I think we could have multiple databases involved. For example you use incremental models in delta but views in iceberg.

Reply:
How others databases could be involved? I'm thinking that in dbt-trino we can configure just one target connector.

Reply:
See https://docs.getdbt.com/docs/building-a-dbt-project/building-models/using-custom-databases

Reply:
Apparently this is not covered by Snowflake as well.

Reply:
@findinpath what's your opinion on supporting multiple catalogs (databases) here. Looks like other adapters don't support it.

Reply:
I have tested locally your implementation and I am not sure whether the current implementation does indeed follow the functionality described on dbt website:

https://docs.getdbt.com/reference/artifacts/catalog-json

> This file contains information from your [data warehouse](https://docs.getdbt.com/terms/data-warehouse) about the tables and views produced and defined by the resources in your project.

Please note the part **resources in your project**.

Now if the generated `catalog.json` contains hundreds/thousands of unrelated tables to the `dbt` project I'm not sure whether is what we want.

I'd naively recommend gathering all the models from the project  and select only those from the information schema.

Note however that this PR is about adding table and column comments. The rest can be done in a follow-up PR.

Reply:
See the [implementation for snowflake](https://github.com/dbt-labs/dbt-snowflake/blob/main/dbt/include/snowflake/macros/catalog.sql):  It seems also not to follow that train of thought.

You could potentially retrieve the full graph of dbt entites through following api https://docs.getdbt.com/reference/dbt-jinja-functions/graph

Let's create a follow-up issue for this and keep this out of scope for this PR

Reply:
I tested graph context variable, and it looks like it's feasible to get all dbt models, not sure about seeds though. 

Sample macro to get all models:
```
{% macro get_models() %}
  {% for node in graph.nodes.values()
     | selectattr("resource_type", "equalto", "model") %}

    {% do log(node.unique_id ~ ", materialized: " ~ node.config.materialized, info=true) %}

  {% endfor %}
{% endmacro %}
```

Command: `dbt run-operation get_models`